### PR TITLE
Add support for API request id as per JSONRPC v2 specs

### PR DESCRIPTION
**Pull request by gianlucastella1**
Issue #1330

### CODE REVIEW

File: libapicore/ApiServer.cpp [hunk 1]
```diff
@@ -153,6 +153,50 @@ static bool checkApiWriteAccess(bool is_read_only, Json::Value& jResponse)
...
 }
 
+static bool parseRequestId(Json::Value& jRequest, Json::Value& jResponse)
+{
+	const char *membername = "id";
+
+	// NOTE: all errors has the same code (-32600) indicating this is an invalid request
+
+	// be sure id is there, otherwise raise an error
+	if (!jRequest.isMember(membername))
```
Comment (AndreaLanfranchi) [inline libapicore/ApiServer.cpp]: Missing or empty can be summarized in a single check ( || )

File: libapicore/ApiServer.cpp [hunk 2]
```diff
@@ -153,6 +153,50 @@ static bool checkApiWriteAccess(bool is_read_only, Json::Value& jResponse)
     return !is_read_only;
 }
 
+static bool parseRequestId(Json::Value& jRequest, Json::Value& jResponse)
+{
+    const char *membername = "id";
+
+    // NOTE: all errors has the same code (-32600) indicating this is an invalid request
```
Comment (XhmikosR) [inline libapicore/ApiServer.cpp]: Has -> have

File: libapicore/ApiServer.cpp [hunk 3]
```diff
@@ -153,6 +153,50 @@ static bool checkApiWriteAccess(bool is_read_only, Json::Value& jResponse)
...
+    if (!jRequest.isMember(membername) || jRequest[membername].empty())
+    {
+        jResponse[membername] = Json::nullValue;
+        jResponse["error"]["code"] = -32600;
+        jResponse["error"]["message"] = "Invalid Request (missing or empty id)";
+        return false;
+    }
+
+    // check if it's empty
+    if (jRequest[membername].empty())
```
Comment (AndreaLanfranchi) [inline libapicore/ApiServer.cpp]: No need to do the same test if you already checked for empty few lines above.
